### PR TITLE
GG-34546 Fixed an issue that failed deactivation of cluster

### PR DESCRIPTION
**Pull request by sk0x50**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java [hunk 1]
```diff
@@ -2184,10 +2186,81 @@ public static ClusterState clusterStateWithLessFeatures(ClusterState state1, Clu
...
+
         return ctx.cache().cacheDescriptors().values().stream()
-            .filter(desc -> !isPersistentCache(desc.cacheConfiguration(), ctx.config().getDataStorageConfiguration())
-                && (!desc.cacheConfiguration().isWriteThrough() || !desc.cacheConfiguration().isReadThrough())
-                && !CU.isSystemCache(desc.cacheName())).map(DynamicCacheDescriptor::cacheName).collect(Collectors.toList());
+            .filter(desc -> !CU.isSystemCache(desc.cacheName()))
+            .filter(desc -> {
+                String dataRegionName = desc.cacheConfiguration().getDataRegionName();
+
+//                log.warning(">>>>> ctx=" + (ctx!=null));
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java:2238]: Please remove these comments

**Review by ibessonov — APPROVED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java [hunk 2]
```diff
@@ -1987,6 +1987,52 @@ public static boolean isInMemoryCluster(Collection<ClusterNode> nodes, JdkMarsha
...
+    /**
+     * Finds and returns a data region configuration with the specified name.
+     *
+     * @param dsCfg Data storage configuration.
+     * @param name Name of data region configuration to find.
+     * @return Data region configuration with the specified name
+     *          or {@code null} if the given data storage configuration does not cantain such data region.
+     */
+    @Nullable public static DataRegionConfiguration findDataRegionConfiguration(
+        DataStorageConfiguration dsCfg,
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java:1999]: Please add @Nullable for arguments.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java [hunk 3]
```diff
@@ -1987,6 +1987,52 @@ public static boolean isInMemoryCluster(Collection<ClusterNode> nodes, JdkMarsha
...
+        if (dsCfg == null || name == null)
+            return null;
+
+        if (dsCfg.getDefaultDataRegionConfiguration().getName().equals(name))
+            return dsCfg.getDefaultDataRegionConfiguration();
+
+        if (dsCfg.getDataRegionConfigurations() == null)
+            return null;
+
+        return Arrays.stream(dsCfg.getDataRegionConfigurations())
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java:2011]: Maybe better to use simple cycle?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java [hunk 4]
```diff
@@ -1987,6 +1987,52 @@ public static boolean isInMemoryCluster(Collection<ClusterNode> nodes, JdkMarsha
...
+     * @param clsLdr Classloader  that is used in order to extract data storage configuration.
+     * @param name Name of data region configuration to find.
+     * @return Data region configuration with the specified name
+     *          or {@code null} if the given data storage configuration does not cantain such data region.
+     */
+    @Nullable public static DataRegionConfiguration findRemoteDataRegionConfiguration(
+        ClusterNode node,
+        JdkMarshaller marshaller,
+        ClassLoader clsLdr,
+        String name
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java:2031]: Please add @nullable for **Srting name**

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java (hunk 1 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java]: Fixed. Thank you for pointing this out.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java (hunk 2 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java]: Ok, fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java (hunk 3 above)
Comment (sk0x50) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheUtils.java]: Yep, it would be better.

**Pull request closed without merging**